### PR TITLE
Figure.solar: Add type hints to the 'terminator' parameter and simplify codes

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

**Preview**: https://pygmt-dev--2881.org.readthedocs.build/en/2881/api/generated/pygmt.Figure.solar.html

Address #2812.

### CODE REVIEW

File: pygmt/src/solar.py [hunk 1]
```diff
@@ -22,7 +26,12 @@
     t="transparency",
 )
 @kwargs_to_strings(R="sequence", c="sequence_comma", p="sequence")
-def solar(self, terminator="d", terminator_datetime=None, **kwargs):
+def solar(
+    self,
+    terminator: Literal["day_night", "civil", "nautical", "astronomical"] = "day_night",
+    terminator_datetime=None,
+    **kwargs,
+):
```
Comment (yvonnefroehlich) [inline pygmt/src/solar.py:34]: Maybe we want to sort the types of terminators alphabetically?
```suggestion
def solar(
    self,
    terminator: Literal["astronomical","civil","day_night","nautical"] = "day_night",
    terminator_datetime=None,
    **kwargs,
):
```

File: pygmt/src/solar.py [hunk 2]
```diff
@@ -36,11 +45,11 @@ def solar(self, terminator="d", terminator_datetime=None, **kwargs):
 
     Parameters
     ----------
-    terminator : str
+    terminator
         Set the type of terminator displayed. Valid arguments are
         ``"day_night"``, ``"civil"``, ``"nautical"``, and ``"astronomical"``,
         which can be set with either the full name or the first letter of the
```
Comment (yvonnefroehlich) [inline pygmt/src/solar.py:51]: Now the available types of terminators are already documented within the function string. So we can maybe shorten this docstring to:

```suggestion
    terminator
        Set the type of terminator displayed, which can be set with either
        the full name or the first letter of the name.
```

File: pygmt/src/solar.py [hunk 3]
```diff
@@ -36,11 +45,11 @@ def solar(self, terminator="d", terminator_datetime=None, **kwargs):
...
-        name [Default is ``"day_night"``].
+        name.
```
Comment (yvonnefroehlich) [inline pygmt/src/solar.py:52]:
```suggestion
```

File: pygmt/src/solar.py (hunk 2 above)
Comment (seisman) [reply pygmt/src/solar.py]: Similar to the `data_source` parameter of the `load_earth_relief` function, I've written the description to:

```
   terminator
        Set the type of terminator displayed, which can be set with either the
        full name or the first letter of the name. Available options are:
        - ``"astronomical"``: Astronomical twilight
        - ``"civil"``: Civil twilight
        - ``"day_night"``: Day/night terminator
        - ``"nautical"``: Nautical twilight
        Refer to https://en.wikipedia.org/wiki/Twilight for the definitions of
        different types of twilight.
```

**Pull request merged**